src/deploy/functions/cache/applyHash.ts

```diff
@@ -0,0 +1,31 @@
+import * as backend from "../backend";
+import * as args from "../args";
+
+/**
+ *
+ */
+export async function applyBackendHashToBackends(
+  wantBackends: Record<string, backend.Backend>,
+  context: args.Context
+) {
+  for (const [codebase, wantBackend] of Object.entries(wantBackends)) {
+    const source = context?.sources?.[codebase];
+    await applyBackendHashToBackend(wantBackend, source);
+  }
+}
+
+/**
+ *
+ */
+export async function applyBackendHashToBackend(wantBackend: backend.Backend, source?: args.Source) {
+  // Blocked by #4852
+  // const hash = await getBackendHash(wantBackend: backend.Backend, source?: args.Source);
+  const hash = "Hash" + wantBackend + source;
+  backend.allEndpoints(wantBackend).forEach((endpoint) => {
+    // Blocked by #4866
+    // endpoint.hash = hash;
```

Comment:
Interesting. I hadn't thought about making `hash` a separate field that's redundant with the label. I don't know if it's necessary. I don't know if it hurts anything.

Reply:
It is a little redundant, but I am a fan of the separate field for 3 reasons:

1. Documenting the field.
2. Only relying on the behavior of `label`'s `Record<string, string>` feels brittle and error prone. Theres a few points we re-assign the labels during the `release` phase. Although we spread the existing labels (so nothing is "lost"), I want to avoid important fields getting unintentionally clobbered.
3. Prior similar art of `Endpoint.codebase`

Reply:
All sound like pretty compelling reasons. Thanks!